### PR TITLE
Create funnelback template

### PR DESCRIPTION
**Pull request by RichardAllum**

### CODE REVIEW

File: course-search.ftl [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+
+See also:
+http://docs.funnelback.com/ui_modern_form_content_type_collection_cfg.html
+
+-->
+    <@fb.ViewModeBanner />
+    <@fb.ErrorMessage />
+    <@s.InitialFormOnly>
+    </@s.InitialFormOnly>
```
Comment (ncn1) [inline course-search.ftl:17]:
```suggestion
```

File: course-search.ftl [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    <@s.AfterSearchOnly>
```
Comment (ncn1) [inline course-search.ftl:18]:
```suggestion
    <@s.AfterSearchOnly>
    <#-- ie. conditional display, content is evaluated only when there are results. -->
```

File: course-search.ftl [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            {
+            "results": [
+            <@s.Results>
+                <#if s.result.class.simpleName == "TierBar">
+                    <#if s.result.matched == s.result.outOf>
+                    <#else></#if>
+                <#else>
+                <#-- EACH RESULT -->
+                    {
+                    "title": "${s.result.metadata.courseTitle}",
```
Comment (ncn1) [inline course-search.ftl:41]:
```suggestion
                    "title": "${s.result.metaData.courseTitle}",
```

File: course-search.ftl [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            "status":"ERROR",
+            "errorMessage":"No matching results."
+            }
+            }
+        <#else>
+        <#-- RESULTS -->
+            {
+            "results": [
+            <@s.Results>
+                <#if s.result.class.simpleName == "TierBar">
```
[the remark this answers is not in the document]
Comment (ncn1) [reply course-search.ftl:35]: 'Results are grouped by tier bars depending of how they match the query terms. There is usually one tier bar for the fully matching results and another one for the partially matching results.'
Quote from https://docs.funnelback.com/modernui-datamodel/com/funnelback/publicui/search/model/padre/TierBar.html
It is necessary, as removing it causes an error.

File: course-search.ftl [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                    <#if s.result.matched == s.result.outOf>
+                    <#else></#if>
```
Comment (ncn1) [inline course-search.ftl:37]:
```suggestion
```

File: course-search.ftl [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+<#compress>
+<#-- course-search.ftl
+
+Outputs Funnelback response in JSON format, with callback if 'callback' parameter is included in query string
+
+See also:
+http://docs.funnelback.com/ui_modern_form_content_type_collection_cfg.html
+
+-->
+    <@fb.ViewModeBanner />
```
Comment (ncn1) [inline course-search.ftl:13]:
```suggestion
    <@fb.ViewModeBanner />
```

File: course-search.ftl [hunk 7]
```diff
@@ -6,57 +6,39 @@
...
-                    }<#if s.result.rank &lt; response.resultPacket.resultsSummary.currEnd>,</#if>
-                </#if>
-            </@s.Results>
-            ]
-            }
-        </#if>
+        {
+        "results": [
+        <@s.Results>
+            <#if s.result.class.simpleName == "TierBar">
```
Comment (FergusMcGlynn) [inline course-search.ftl:22]: Could this be `if simpleName != "TierBar"` and then drop the `else`? Also I still don't really understand what "TierBar" means, so an inline comment might help to clarify?

**Review by FergusMcGlynn — CHANGES_REQUESTED** [review]

**Review by ncn1 — APPROVED** [review]

**Review by FergusMcGlynn — APPROVED** [review]

**Pull request merged**